echopype/consolidate/loc_utils.py

```diff
@@ -0,0 +1,162 @@
+                        f"latitude_{potential_datagram_type.lower()}"
+                    )
+                elif potential_datagram_type is None:
+                    potential_lat_var = echodata["Platform"].get("latitude")
+                    potential_lon_var = echodata["Platform"].get("longitude")
+
+                # Check if loc vars are invalid
+                if validity_check in ["missing", "all_nan"]:
+                    potential_invalid_bool = compute_invalid_check(
+                        potential_lat_var, potential_lon_var, "missing"
+                    ) or compute_invalid_check(potential_lat_var, potential_lon_var, "all_nan")
+                elif validity_check in ["some_nan", "some_zero"]:
+                    potential_invalid_bool = compute_invalid_check(
+                        potential_lat_var, potential_lon_var, "some_nan"
+                    ) or compute_invalid_check(potential_lat_var, potential_lon_var, "some_zero")
+                # Append to good datagram types
+                if not potential_invalid_bool:
+                    good_datagram_types.append(potential_datagram_type)
+
+            if len(good_datagram_types) > 0:
+                # Append to output message
+                output_message = (
+                    output_message
+                    + f" Consider setting datagram_type to any of {good_datagram_types}."
+                )
+
+        # Raise error / warning depending on validity_check value
+        if validity_check in ["missing", "all_nan"]:
+            raise ValueError(output_message)
+        elif validity_check in ["some_nan", "some_zero"]:
+            logger.warning(output_message)
+
+
+def check_loc_time_dim_duplicates(echodata: EchoData, time_dim_name: str) -> None:
+    """Check if there are duplicates in time_dim_name"""
+    if len(np.unique(echodata["Platform"][time_dim_name].data)) != len(
+        echodata["Platform"][time_dim_name].data
+    ):
+        raise ValueError(
+            f'The ``echodata["Platform"]["{time_dim_name}"]`` array contains duplicate values. '
+            "Downstream interpolation on the position variables requires unique time values."
+        )
+
+
+def sel_interp(
+    ds: xr.Dataset,
+    echodata: EchoData,
+    datagram_type: str,
+    loc_name: str,
+    time_dim_name: str,
+    nmea_sentence: str,
+) -> xr.DataArray:
+    """
+    Selection and interpolation for a location variable.
+
+    The selection logic is as follows, with 4 possible scenarios:
+
+    1) If datagram_type == NMEA is used and NMEA sentence is NaN, then do nothing.
+    2) If datagram_type == NMEA is used and NMEA sentence is not NaN, then do the selection.
+    3) If datagram_type != NMEA and NMEA sentence is NaN, then do nothing.
+    4) If datagram_type != NMEA and NMEA sentence is not NaN, then raise ValueError since NMEA
+    sentence selection can only be used on location variables from the NMEA datagram.
```

Comment:
```suggestion
    The selection logic is as follows, with 4 possible scenarios:
    Note here datagram_type = None is equivalent to datagram_type = NMEA
    
    1) If datagram_type is None and nmea_sentence is None, then do nothing.
    2) If datagram_type is None and nmea_sentence is not None, then do the selection.
    3) If datagram_type is not None and nmea_sentence is None, then do nothing.
    4) If datagram_type is not None and nmea_sentence is not None, then raise ValueError since NMEA
       sentence selection can only be used on location variables from the NMEA datagram.
```